### PR TITLE
Feature/allow gk code to be set when writing

### PR DESCRIPTION
**Pull request by d7919**
Quality of life feature to make it easier to transiently set the gk_code type for output.

### CODE REVIEW

File: pyrokinetics/pyro.py [hunk 1]
```diff
@@ -222,6 +229,9 @@ def write_gk_file(self,
 
         self.gk_code.write(self, file_name, directory=self.run_directory)
 
+        # Ensure that gk_code is unchanged on exit
+        self.gk_code = original_gk_code
```
Comment (bpatel2107) [inline pyrokinetics/pyro.py:233]: @d7919  Not sure if it is necessary to save the original gk_code. No data is lost by changing the code as the namelist data for each code is saved under a separate object.

For example the if we load in a GS2 input file and the namelist data with all the flags is saved under pyro.gs2_input. When changing code the pyro.gs2_input remains. Thoughts?

Comment (d7919) [reply pyrokinetics/pyro.py:233]: I prefer function calls to not have side effects if possible. The original approach makes it clear that the user is changing the gk_code value of the pyro instance whilst this function call doesn't make it obvious so I thought we should revert to the original type.

Comment (bpatel2107) [reply pyrokinetics/pyro.py:233]: Fair enough, I think I agree with that

Comment (bpatel2107) [reply pyrokinetics/pyro.py:233]: This currently won't quite work as you can only set gk_code with a string. We need modify the gk_code.setter to handle gk_code objects as inputs

Comment (bpatel2107) [reply pyrokinetics/pyro.py:233]: That would work but we definitely need to re-vist how setting gk_code/gk_type works

**Review by bpatel2107 — APPROVED** [review]

**Pull request merged**